magento/magento2
# magento/magento2#23054

### Description (*)

Extending Magento/Cron module with functionality of being able to close the orphan/stale running cron tasks (schedules) which been started before, but somehow stopped running

### Fixed Issues (if relevant)

1. Fixes #23054 : Cron job not running after crashed once

### Manual testing scenarios (*)

preconditions:
1. _ps_ command availability on server
2. _php bin/magento setup:upgrade_ to apply modifications in database
3. run _php bin/magento cron:run_ multiple time (or be sure that crontab does the same) to get the necessary data for process schedules

(scenario a)
1. change "success" to "running" on couple of job on _cron_schedule_ table
2. run _php bin/magento cron:run_ 
3. check the modified rows

(scenario b)
1. run _php bin/magento cron:run_ on CLI
2. shut down/restart the server from a different console before getting back the prompt
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/set-hostname.html
### Questions or comments

If you using multiple instances or in cloud environment, be sure that you have set up the hostname(s) of your instance(s) (eg. https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/set-hostname.html) in prior to avoid false negative match if you checking the hostnames as well

### Contribution checklist (*)
